extensions/podman/packages/extension/src/extension.ts

```diff
@@ -177,6 +196,43 @@ function notifySetupPodman(): void {
   notificationDisposable = extensionApi.window.showNotification(setupPodmanNotification);
 }
 
+async function checkAndNotifySetupPodmanMacHelper(): Promise<void> {
```

Comment:
question: is there a way to move this logic to a dedicated class? like something on its own, the `extension.ts` file is 2k longs already 😿 

- getSocketCompatibility is from utils
- isDisguisedPodman is from utils

The notification could be managed by the class, the class could implement Disposable, and instead of adding 3 functions we add one class import and constructor

Reply:
I could do that in a separate PR, as I'd have to move `setupPodmanNotification` as well since that should be in a class too.

Reply:
please create the issue and add it as a sub-issue of https://github.com/podman-desktop/podman-desktop/issues/10327, so we don't lose track of it :+1:

Reply:
Done! https://github.com/podman-desktop/podman-desktop/issues/12064 @axel7083